### PR TITLE
Image Tagging in Discovery

### PR DESCRIPTION
**Pull request by ghost**
https://cognitedata.atlassian.net/browse/CONTEXT-308

image

Feel free to use `contextualization` tenant

### CODE REVIEW

File: src/utils/detectionApi.ts [hunk 1]
```diff
@@ -0,0 +1,506 @@
...
+    const metadata = this.metadataMergeInto(metadatamapped, detection.metadata);
+
+    return {
+      externalId: detection.externalId,
+      startTime: detection.startTime,
+      endTime: detection.endTime,
+      type: this.EVENT_TYPE,
+      subtype: detection.label,
+      description: detection.description,
+      metadata,
```
Comment (miladcognite) [inline src/utils/detectionApi.ts]: Should it be this?

```suggestion
      metadata: metadata,
```

File: src/utils/detectionApi.ts [hunk 2]
```diff
@@ -0,0 +1,506 @@
...
+   */
+  label?: string;
+  /**
+   * Filter detections with an externalId starting with this value
+   */
+  externalIdPrefix?: string;
+  /**
+   * Filter based on file id
+   */
+  fileId?: string;
```
Comment (miladcognite) [inline src/utils/detectionApi.ts]: Should this be `fileExternalId`? (cc: @runevalle-cognite )

File: src/utils/detectionApi.ts (hunk 1 above)
Comment (dlcognite) [reply src/utils/detectionApi.ts]: ahh the linting would short hand this :)

File: src/utils/detectionApi.ts (hunk 2 above)
Comment (dlcognite) [reply src/utils/detectionApi.ts]: i added this
so the current logic is use externalId if exist, if not, then internalId

Comment (dlcognite) [reply src/utils/detectionApi.ts]: i can change it to be fileExternalId instead :)

**Review by miladcognite — APPROVED** [review]
Very nice! :+1:

**Review by anderscognite — APPROVED** [review]
LGTM

**Pull request merged**